engine_redis.go

```diff
@@ -735,54 +735,103 @@ func (s *shard) useCluster() bool {
 }
 
 func (s *shard) controlChannelID() channelID {
-	return channelID(s.config.Prefix + redisControlChannelSuffix)
+	return channelID(strings.Join([]string{s.config.Prefix, redisControlChannelSuffix}, ""))
```

Comment:
can not accept this without benchmark, I am almost sure we are losing performance here since Go has optimization for string concatenation using plus sign.